Access Token Cookie expiry date is 1 second off of exp claim in Access Token

Note: Spring WebMVC and Servlet do NOT exhibit this issue. Spring Boot WebMVC DOES exhibit this issue.

Steps to reproduce:

1) Run the Spring WebMVC example from the Java SDK project.
2) Run the `CookieIT.cookieExpirationMatchesTokenTtl` test from the `stormpath-framework-tck`

    Result:
    ===============================================
    Default Suite
    Total tests run: 1, Failures: 0, Skips: 0
    ===============================================

3) Run the Spring Boot WebMVC example from the Java SDK project.
4) Run the `CookieIT.cookieExpirationMatchesTokenTtl` test from the `stormpath-framework-tck`

    Result:
    java.lang.AssertionError: expected [Tue Jul 19 19:31:05 EDT 2016] but found [Tue Jul 19 19:31:06 EDT 2016]
    Expected :Tue Jul 19 19:31:05 EDT 2016
    Actual   :Tue Jul 19 19:31:06 EDT 2016
    
    at com.stormpath.tck.authentication.CookieIT.cookieExpirationMatchesTokenTtl(CookieIT.groovy:126)

I don't know if this helps, but when I run it, it's off by 15 seconds:

```
java.lang.AssertionError: expected [Mon Jul 25 16:43:27 MDT 2016] but found [Mon Jul 25 16:43:42 MDT 2016]
Expected :Mon Jul 25 16:43:27 MDT 2016
Actual   :Mon Jul 25 16:43:42 MDT 2016
```

--- a/extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/account/CookieAuthenticationResultSaver.java
+++ b/extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/account/CookieAuthenticationResultSaver.java
@@ -94,7 +94,10 @@ public class CookieAuthenticationResultSaver implements Saver<AuthenticationResu
             String refreshToken = accessTokenResult.getTokenResponse().getRefreshToken();
 
             getAccessTokenCookieSaver(request, getMaxAge(accessToken, clientSecret)).set(request, response, accessToken);
-            getRefreshTokenCookieSaver(request, getMaxAge(refreshToken, clientSecret)).set(request, response, refreshToken);
+            //Client Credentials auth workflow doesn't contains a refresh token
+            if (Strings.hasText(refreshToken)) {
+                getRefreshTokenCookieSaver(request, getMaxAge(refreshToken, clientSecret)).set(request, response, refreshToken);
+            }
         }
         if (value instanceof TransientAuthenticationResult) {
             Account account = value.getAccount();

